File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/productsentinels/domain/ProductSentinelsObject.java
package com.zwstudio.logicpuzzlesandroid.puzzles.productsentinels.domain;


public abstract class ProductSentinelsObject {
    public abstract String objAsString();
    public static ProductSentinelsObject objFromString(String str) {
        switch (str) {
            case "marker":
                return new ProductSentinelsMarkerObject();
            case "tower":
                return new ProductSentinelsTowerObject();
            default:
                return new ProductSentinelsMarkerObject();
        }
    }
}

File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/productsentinels/ProductSentinelsObject.kt
package com.zwstudio.logicpuzzlesandroid.puzzles.productsentinels

import com.zwstudio.logicpuzzlesandroid.common.domain.Position

enum class ProductSentinelsObject {
    Empty, Forbidden, Hint, Marker, Tower
}

class ProductSentinelsGameMove(val p: Position, var obj: ProductSentinelsObject = ProductSentinelsObject.Empty)
